module-system/sov-modules-api/src/lib.rs

```diff
@@ -235,6 +239,18 @@ pub trait Module {
     }
 }
 
+/// A [`Module`] that has a well-defined and known [JSON
+/// Schema](https://json-schema.org/) for its [`Module::CallMessage`].
+///
+/// This trait is intended to support code generation tools, CLIs, and
+/// documentation. You can derive it with `#[derive(ModuleCallJsonSchema)]`, or
+/// implement it manually if your use case demands more control over the JSON
+/// Schema generation.
+pub trait ModuleCallJsonSchema: Module {
```

Comment:
This trait looks good, but I think we also want Json Schemas for RPC query messages. I don't have a strong opinion about whether we should have two separate traits (one for call messages and one for queries), or package them both into one. One argument in favor of two traits is that we may have additional `derives` we want to package together for call messages (`BorshSchema`, `BorshSerialize`, `clap::Parser` etc), while query messages probably only need `serde` and `JsonSchema` to support RPC.

cc @bkolad and @dubbelosix - any thoughts?

Reply:
I don't have any strong opinion here. I am fine with postponing the decision until we start working on JSON Schemas for  RPC queries and see how it looks.

Reply:
I don't have any strong opinion here. I am fine with postponing the decision until we start working on JSON Schemas for RPC queries and see how it looks.

Reply:
That's fine by me 👍